README.rst
======================
django-nomad-i18n-blog
======================

Overview
========

Django reusable app to implement a system of
country-specific blogs (speficied by its ISO code).


CHANGELOG.rst
=========
CHANGELOG
=========

v4.2.0
======

* Feature posts in country blogs

v4.1.1
======

* Add country_code in template context

v4.1.0
======

* Rename urlpatterns and templates

v4.0.0
======

* Refactor model schema to allow easier relationships and structure.

v3.0.0
======

* Categories are now unique and shared accross blog posts

v2.1.4
======

* Remove unique constraint in Category.slug

v2.1.3
======

* Remove locale files from package

v2.1.2
======

* Add missing locale files to package

v2.1.1
======

* Use lazy translations on models

v2.1.0
======

* Categories must be related to a blog
* Remove migrations, use SOUTH_MIGRATION_MODULES instead

v2.0.1
======

* Fix validation over object updates

v2.0.0
======

* Improve models, add i18n

v1.0.0
======

* Add blog hubs so blogs can be grouped into communities

v0.9.1
======

* First release


nomadblog/admin.py
from django.contrib import admin

from nomadblog.models import BlogHub, Blog, Category, BlogUser


class BlogHubAdmin(admin.ModelAdmin):
    list_display = ('name', )
    search_fields = ('name', )


class BlogAdmin(admin.ModelAdmin):
    list_display = ('title', 'country_code', 'slug', 'description')
    search_fields = ('title', 'slug', 'description')
    list_filter = ('country_code', )


class BlogUserAdmin(admin.ModelAdmin):
    list_display = ('user', 'blog', 'bio')
    search_fields = ('user__username', 'user__email', 'blog__title', 'blog__country_code', 'bio')


class CategoryAdmin(admin.ModelAdmin):
    list_display = ('name', 'slug')
    search_fields = ('name', 'slug')


admin.site.register(BlogHub, BlogHubAdmin)
admin.site.register(Blog, BlogAdmin)
admin.site.register(Category, CategoryAdmin)
admin.site.register(BlogUser, BlogUserAdmin)


nomadblog/views.py
from django.views.generic import ListView, DetailView
from django.shortcuts import get_object_or_404
from django.conf import settings

from nomadblog.models import Blog, Category
from nomadblog import get_post_model


DEFAULT_STATUS = getattr(settings, 'PUBLIC_STATUS', 0)
POST_MODEL = get_post_model()


class NomadBlogMixin(object):

    def dispatch(self, request, *args, **kwargs):
        self.blog = get_object_or_404(Blog, countries__code__iexact=self.kwargs.get('country_code'), slug=self.kwargs.get('blog_slug'))
        self.country_code = self.kwargs.get('country_code')
        return super(NomadBlogMixin, self).dispatch(request, *args, **kwargs)

    def get_context_data(self, *args, **kwargs):
        context = super(NomadBlogMixin, self).get_context_data(*args, **kwargs)
        context['blog'] = self.blog
        context['country_code'] = self.country_code
        return context


class PostList(NomadBlogMixin, ListView):
    model = POST_MODEL
    template_name = 'nomadblog/post_list.html'

    def get_queryset(self):
        qs = super(PostList, self).get_queryset()
        return qs.filter(bloguser__blog=self.blog)


class PostDetail(NomadBlogMixin, DetailView):
    model = POST_MODEL
    template_name = 'nomadblog/post_detail.html'

    def get_object(self, queryset=None):
        queryset = self.get_queryset().filter(bloguser__blog=self.blog)
        return super(PostDetail, self).get_object(queryset)


class CategoriesList(NomadBlogMixin, ListView):
    model = Category


class PostsByCategoryList(NomadBlogMixin, ListView):
    model = POST_MODEL
    template_name = 'nomadblog/post_list_by_category.html'

    def get_queryset(self, *args, **kwargs):
        qs = super(PostsByCategoryList, self).get_queryset()
        self.category = get_object_or_404(Category, slug=self.kwargs.get('category_slug', ''))
        return qs.filter(categories=self.category)

    def get_context_data(self, *args, **kwargs):
        context = super(PostsByCategoryList, self).get_context_data(*args, **kwargs)
        context['category'] = self.category
        return context


nomadblog/models.py
from django.db import models
from django.conf import settings
from django.core.urlresolvers import reverse
from django.utils.translation import ugettext_lazy as _
from django.core.validators import ValidationError
from django.core.exceptions import NON_FIELD_ERRORS
from django.core.exceptions import ObjectDoesNotExist


class Country(models.Model):
    code = models.CharField(_('iso country code'), max_length=2, help_text='Reference: http://www.iso.org/iso/home/standards/country_codes/country_names_and_code_elements_txt.htm')
    name = models.CharField(_('name'), max_length=100)

    def __unicode__(self):
        return u"%s" % self.name

    class Meta:
        verbose_name = _('country')
        verbose_name_plural = _('countries')


class BlogHub(models.Model):
    name = models.CharField(_('blog hub'), max_length=100)

    def __unicode__(self):
        return u"%s" % self.name

    class Meta:
        verbose_name = _('blog hub')
        verbose_name_plural = _('blog hubs')


class Blog(models.Model):
    title = models.CharField(_('title'), max_length=100)
    slug = models.SlugField(_('slug'), max_length=50, unique=True)
    description = models.TextField(_('description'))
    creation_date = models.DateTimeField(_('creation date'), auto_now_add=True)
    countries = models.ManyToManyField(Country, verbose_name=_('countries'))
    users = models.ManyToManyField(settings.AUTH_USER_MODEL, through='BlogUser', verbose_name=_('users'))
    hubs = models.ManyToManyField(BlogHub, verbose_name=_('hubs'))
    seo_title = models.CharField(_('seo title'), max_length=70, blank=True)
    seo_desc = models.CharField(_('seo meta description'), max_length=160, blank=True)

    class Meta:
        verbose_name = _('blog')
        verbose_name_plural = _('blogs')

    def __unicode__(self):
        return u"%s" % self.title


class BlogUser(models.Model):
    user = models.ForeignKey(settings.AUTH_USER_MODEL, verbose_name=_('user'))
    slug = models.SlugField(_('slug'), max_length=50)
    blog = models.ForeignKey(Blog, verbose_name=_('blog'))
    bio = models.CharField(_('bio'), max_length=255, blank=True)
    seo_title = models.CharField(_('seo title'), max_length=70, blank=True)
    seo_desc = models.CharField(_('seo meta description'), max_length=160, blank=True)

    def __unicode__(self):
        return u"%s - %s" % (self.user.username, self.blog)

    class Meta:
        unique_together = ('slug', 'blog')
        verbose_name = _('blog user')
        verbose_name_plural = _('blog users')


class Category(models.Model):
    name = models.CharField(_('name'), max_length=100)
    slug = models.SlugField(_('slug'), max_length=50, unique=True)
    description = models.CharField(_('description'), max_length=500)
    seo_title = models.CharField(_('seo title'), max_length=70, blank=True)
    seo_desc = models.CharField(_('seo meta description'), max_length=160, blank=True)

    def __unicode__(self):
        return u"%s" % self.name

    class Meta:
        verbose_name = _('category')
        verbose_name_plural = _('categories')


class Post(models.Model):
    PUBLIC_STATUS = 0
    DRAFT_STATUS = 1
    PRIVATE_STATUS = 2
    DEFAULT_POST_STATUS_CHOICES = (
        (PUBLIC_STATUS, 'public'),
        (DRAFT_STATUS, 'draft'),
        (PRIVATE_STATUS, 'private'),
    )
    POST_STATUS_CHOICES = getattr(settings, 'POST_STATUS_CHOICES', DEFAULT_POST_STATUS_CHOICES)

    bloguser = models.ForeignKey(BlogUser, verbose_name=_('blog user'))
    title = models.CharField(_('title'), max_length=300)
    slug = models.SlugField(_('slug'), max_length=50)
    pub_date = models.DateTimeField(_('publication date'), auto_now_add=True)
    status = models.IntegerField(_('status'), choices=POST_STATUS_CHOICES, default=0)
    categories = models.ManyToManyField(Category, verbose_name=_('categories'))
    content = models.TextField(_('content'))
    seo_title = models.CharField(_('seo title'), max_length=70, blank=True)
    seo_desc = models.CharField(_('seo meta description'), max_length=160, blank=True)

    def __unicode__(self):
        return u"%s - %s" % (self.bloguser, self.title)

    class Meta:
        abstract = True
        unique_together = ('bloguser', 'slug')


nomadblog/migrations/0002_auto__add_bloghub.py
# -*- coding: utf-8 -*-
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


# Safe User import for Django < 1.5
try:
    from django.contrib.auth import get_user_model
except ImportError:
    from django.contrib.auth.models import User
else:
    User = get_user_model()

# With the default User model these will be 'auth.User' and 'auth.user'
# so instead of using orm['auth.User'] we can use orm[user_orm_label]
user_orm_label = '%s.%s' % (User._meta.app_label, User._meta.object_name)
user_model_label = '%s.%s' % (User._meta.app_label, User._meta.module_name)


class Migration(SchemaMigration):

    def forwards(self, orm):
        # Adding model 'BlogHub'
        db.create_table(u'nomadblog_bloghub', (
            (u'id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('name', self.gf('django.db.models.fields.CharField')(max_length=100)),
        ))
        db.send_create_signal(u'nomadblog', ['BlogHub'])

        # Adding M2M table for field hubs on 'Blog'
        m2m_table_name = db.shorten_name(u'nomadblog_blog_hubs')
        db.create_table(m2m_table_name, (
            ('id', models.AutoField(verbose_name='ID', primary_key=True, auto_created=True)),
            ('blog', models.ForeignKey(orm[u'nomadblog.blog'], null=False)),
            ('bloghub', models.ForeignKey(orm[u'nomadblog.bloghub'], null=False))
        ))
        db.create_unique(m2m_table_name, ['blog_id', 'bloghub_id'])

    def backwards(self, orm):
        # Deleting model 'BlogHub'
        db.delete_table(u'nomadblog_bloghub')

        # Removing M2M table for field hubs on 'Blog'
        db.delete_table(db.shorten_name(u'nomadblog_blog_hubs'))

    models = {
        user_model_label: {
            'Meta': {
                'object_name': User.__name__,
                'db_table': "'%s'" % User._meta.db_table
            },
            User._meta.pk.attname: (
                'django.db.models.fields.AutoField', [],
                {'primary_key': 'True', 'db_column': "'%s'" % User._meta.pk.column}
            ),
        },
        u'nomadblog.blog': {
            'Meta': {'object_name': 'Blog'},
            'country_code': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '2'}),
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'auto_now_add': 'True', 'blank': 'True'}),
            'description': ('django.db.models.fields.CharField', [], {'max_length': '500'}),
            'hubs': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['nomadblog.BlogHub']", 'symmetrical': 'False'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '50'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'users': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['%s']" % user_orm_label, 'through': u"orm['nomadblog.BlogUser']", 'symmetrical': 'False'})
        },
        u'nomadblog.bloghub': {
            'Meta': {'object_name': 'BlogHub'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        u'nomadblog.bloguser': {
            'Meta': {'object_name': 'BlogUser'},
            'bio': ('django.db.models.fields.CharField', [], {'max_length': '255', 'blank': 'True'}),
            'blog': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['nomadblog.Blog']"}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['%s']" % user_orm_label})
        },
        u'nomadblog.category': {
            'Meta': {'object_name': 'Category'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'slug': ('django.db.models.fields.SlugField', [], {'unique': 'True', 'max_length': '50'})
        }
    }

    complete_apps = ['nomadblog']


nomadblog/migrations/0003_auto__del_unique_blog_country_code__add_unique_blog_country_code_slug.py
# -*- coding: utf-8 -*-
from south.db import db
from south.v2 import SchemaMigration


# Safe User import for Django < 1.5
try:
    from django.contrib.auth import get_user_model
except ImportError:
    from django.contrib.auth.models import User
else:
    User = get_user_model()

# With the default User model these will be 'auth.User' and 'auth.user'
# so instead of using orm['auth.User'] we can use orm[user_orm_label]
user_orm_label = '%s.%s' % (User._meta.app_label, User._meta.object_name)
user_model_label = '%s.%s' % (User._meta.app_label, User._meta.module_name)


class Migration(SchemaMigration):

    def forwards(self, orm):
        # Removing unique constraint on 'Blog', fields ['country_code']
        db.delete_unique(u'nomadblog_blog', ['country_code'])

        # Adding unique constraint on 'Blog', fields ['country_code', 'slug']
        db.create_unique(u'nomadblog_blog', ['country_code', 'slug'])

    def backwards(self, orm):
        # Removing unique constraint on 'Blog', fields ['country_code', 'slug']
        db.delete_unique(u'nomadblog_blog', ['country_code', 'slug'])

        # Adding unique constraint on 'Blog', fields ['country_code']
        db.create_unique(u'nomadblog_blog', ['country_code'])

    models = {
        user_model_label: {
            'Meta': {
                'object_name': User.__name__,
                'db_table': "'%s'" % User._meta.db_table
            },
            User._meta.pk.attname: (
                'django.db.models.fields.AutoField', [],
                {'primary_key': 'True', 'db_column': "'%s'" % User._meta.pk.column}
            ),
        },
        u'nomadblog.blog': {
            'Meta': {'object_name': 'Blog'},
            'country_code': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '2'}),
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'auto_now_add': 'True', 'blank': 'True'}),
            'description': ('django.db.models.fields.CharField', [], {'max_length': '500'}),
            'hubs': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['nomadblog.BlogHub']", 'symmetrical': 'False'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '50'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'users': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['%s']" % user_orm_label, 'through': u"orm['nomadblog.BlogUser']", 'symmetrical': 'False'})
        },
        u'nomadblog.bloghub': {
            'Meta': {'object_name': 'BlogHub'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        u'nomadblog.bloguser': {
            'Meta': {'object_name': 'BlogUser'},
            'bio': ('django.db.models.fields.CharField', [], {'max_length': '255', 'blank': 'True'}),
            'blog': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['nomadblog.Blog']"}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['%s']" % user_orm_label})
        },
        u'nomadblog.category': {
            'Meta': {'object_name': 'Category'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'slug': ('django.db.models.fields.SlugField', [], {'unique': 'True', 'max_length': '50'})
        }
    }

    complete_apps = ['nomadblog']


nomadblog/migrations/0001_initial.py
# -*- coding: utf-8 -*-
from south.utils import datetime_utils as datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


class Migration(SchemaMigration):

    def forwards(self, orm):
        # Adding model 'BlogHub'
        db.create_table(u'nomadblog_bloghub', (
            (u'id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('name', self.gf('django.db.models.fields.CharField')(max_length=100)),
        ))
        db.send_create_signal(u'nomadblog', ['BlogHub'])

        # Adding model 'Blog'
        db.create_table(u'nomadblog_blog', (
            (u'id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=100)),
            ('slug', self.gf('django.db.models.fields.SlugField')(max_length=50)),
            ('description', self.gf('django.db.models.fields.TextField')()),
            ('country_code', self.gf('django.db.models.fields.CharField')(max_length=2)),
            ('creation_date', self.gf('django.db.models.fields.DateTimeField')(auto_now_add=True, blank=True)),
            ('seo_title', self.gf('django.db.models.fields.CharField')(max_length=70, blank=True)),
            ('seo_desc', self.gf('django.db.models.fields.CharField')(max_length=160, blank=True)),
        ))
        db.send_create_signal(u'nomadblog', ['Blog'])

        # Adding unique constraint on 'Blog', fields ['country_code', 'slug']
        db.create_unique(u'nomadblog_blog', ['country_code', 'slug'])

        # Adding M2M table for field hubs on 'Blog'
        m2m_table_name = db.shorten_name(u'nomadblog_blog_hubs')
        db.create_table(m2m_table_name, (
            ('id', models.AutoField(verbose_name='ID', primary_key=True, auto_created=True)),
            ('blog', models.ForeignKey(orm[u'nomadblog.blog'], null=False)),
            ('bloghub', models.ForeignKey(orm[u'nomadblog.bloghub'], null=False))
        ))
        db.create_unique(m2m_table_name, ['blog_id', 'bloghub_id'])

        # Adding model 'BlogUser'
        db.create_table(u'nomadblog_bloguser', (
            (u'id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('user', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['accounts.User'])),
            ('slug', self.gf('django.db.models.fields.SlugField')(max_length=50)),
            ('blog', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['nomadblog.Blog'])),
            ('bio', self.gf('django.db.models.fields.CharField')(max_length=255, blank=True)),
            ('seo_title', self.gf('django.db.models.fields.CharField')(max_length=70, blank=True)),
            ('seo_desc', self.gf('django.db.models.fields.CharField')(max_length=160, blank=True)),
        ))
        db.send_create_signal(u'nomadblog', ['BlogUser'])

        # Adding model 'Category'
        db.create_table(u'nomadblog_category', (
            (u'id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('name', self.gf('django.db.models.fields.CharField')(max_length=100)),
            ('slug', self.gf('django.db.models.fields.SlugField')(unique=True, max_length=50)),
            ('description', self.gf('django.db.models.fields.CharField')(max_length=500)),
            ('seo_title', self.gf('django.db.models.fields.CharField')(max_length=70, blank=True)),
            ('seo_desc', self.gf('django.db.models.fields.CharField')(max_length=160, blank=True)),
        ))
        db.send_create_signal(u'nomadblog', ['Category'])


    def backwards(self, orm):
        # Removing unique constraint on 'Blog', fields ['country_code', 'slug']
        db.delete_unique(u'nomadblog_blog', ['country_code', 'slug'])

        # Deleting model 'BlogHub'
        db.delete_table(u'nomadblog_bloghub')

        # Deleting model 'Blog'
        db.delete_table(u'nomadblog_blog')

        # Removing M2M table for field hubs on 'Blog'
        db.delete_table(db.shorten_name(u'nomadblog_blog_hubs'))

        # Deleting model 'BlogUser'
        db.delete_table(u'nomadblog_bloguser')

        # Deleting model 'Category'
        db.delete_table(u'nomadblog_category')


    models = {
        u'accounts.user': {
            'Meta': {'object_name': 'User'},
            'country': ('django_countries.fields.CountryField', [], {'default': "'MX'", 'max_length': '2'}),
            'date_joined': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'email': ('django.db.models.fields.EmailField', [], {'max_length': '75', 'blank': 'True'}),
            'first_login': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'first_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'groups': ('django.db.models.fields.related.ManyToManyField', [], {'symmetrical': 'False', 'related_name': "u'user_set'", 'blank': 'True', 'to': u"orm['auth.Group']"}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'image': ('django.db.models.fields.files.ImageField', [], {'max_length': '255', 'null': 'True', 'blank': 'True'}),
            'is_active': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'is_staff': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'is_superuser': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'last_login': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'last_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '255', 'blank': 'True'}),
            'password': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '255'}),
            'user_permissions': ('django.db.models.fields.related.ManyToManyField', [], {'symmetrical': 'False', 'related_name': "u'user_set'", 'blank': 'True', 'to': u"orm['auth.Permission']"}),
            'username': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '30'})
        },
        u'auth.group': {
            'Meta': {'object_name': 'Group'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '80'}),
            'permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'})
        },
        u'auth.permission': {
            'Meta': {'ordering': "(u'content_type__app_label', u'content_type__model', u'codename')", 'unique_together': "((u'content_type', u'codename'),)", 'object_name': 'Permission'},
            'codename': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'content_type': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['contenttypes.ContentType']"}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '50'})
        },
        u'contenttypes.contenttype': {
            'Meta': {'ordering': "('name',)", 'unique_together': "(('app_label', 'model'),)", 'object_name': 'ContentType', 'db_table': "'django_content_type'"},
            'app_label': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'model': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        u'nomadblog.blog': {
            'Meta': {'unique_together': "(('country_code', 'slug'),)", 'object_name': 'Blog'},
            'country_code': ('django.db.models.fields.CharField', [], {'max_length': '2'}),
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'auto_now_add': 'True', 'blank': 'True'}),
            'description': ('django.db.models.fields.TextField', [], {}),
            'hubs': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['nomadblog.BlogHub']", 'symmetrical': 'False'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'seo_desc': ('django.db.models.fields.CharField', [], {'max_length': '160', 'blank': 'True'}),
            'seo_title': ('django.db.models.fields.CharField', [], {'max_length': '70', 'blank': 'True'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '50'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'users': ('django.db.models.fields.related.ManyToManyField', [], {'to': u"orm['accounts.User']", 'through': u"orm['nomadblog.BlogUser']", 'symmetrical': 'False'})
        },
        u'nomadblog.bloghub': {
            'Meta': {'object_name': 'BlogHub'},
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        u'nomadblog.bloguser': {
            'Meta': {'object_name': 'BlogUser'},
            'bio': ('django.db.models.fields.CharField', [], {'max_length': '255', 'blank': 'True'}),
            'blog': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['nomadblog.Blog']"}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'seo_desc': ('django.db.models.fields.CharField', [], {'max_length': '160', 'blank': 'True'}),
            'seo_title': ('django.db.models.fields.CharField', [], {'max_length': '70', 'blank': 'True'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '50'}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['accounts.User']"})
        },
        u'nomadblog.category': {
            'Meta': {'object_name': 'Category'},
            'description': ('django.db.models.fields.CharField', [], {'max_length': '500'}),
            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'seo_desc': ('django.db.models.fields.CharField', [], {'max_length': '160', 'blank': 'True'}),
            'seo_title': ('django.db.models.fields.CharField', [], {'max_length': '70', 'blank': 'True'}),
            'slug': ('django.db.models.fields.SlugField', [], {'unique': 'True', 'max_length': '50'})
        }
    }

    complete_apps = ['nomadblog']